test/server/services/ApplicationDeveloper.test.js

```diff
@@ -100,6 +100,29 @@ describe('services/ApplicationDeveloper', () => {
     expect(stubLogger).to.have.been.calledOnce;
   });
 
+  it('should return reject delete a specific key from applications.api service', () => {
+    // static test vars
+    const mockEnv = 'mock';
+    const mockURL = 'https://mocksite.com';
+    const mockAppId = 'test';
+    const mockKeyId = 'testkey';
+    const mockOauthToken = 'token';
+    const mockKeyType = 'incorrectKeyType';
+    const stubAPIClientHelper = sinon.stub(APIClientHelper, 'getPrivateAPIClient').returns(null);
+    applicationDeveloper.server.baseUrl.mock = mockURL;
+    applicationDeveloper.APIClientHelper = stubAPIClientHelper;
+
+    applicationDeveloper.deleteAPIClient(mockAppId, mockKeyId, mockKeyType, mockOauthToken, mockEnv)
+      .then(rslt =>
+        assert(false, 'Method should error')
+      )
+      .catch(rslt => {
+        expect(rslt.message)
+          .to.be.equal('Incorrect Key Type');
+      }
```

Comment:
Yes this is one way of doing it. However it would be better to use the chai-as-promised methods. It has been setup in test/setup.js. You should be able to do something like

```
const promise = applicationDeveloper.deleteAPIClient(mockAppId, mockKeyId, mockKeyType, mockOauthToken, mockEnv)
return expect(promise).should.be.rejectedWith(Error)
```

https://www.chaijs.com/plugins/chai-as-promised/

Reply:
That makes life easier - thanks will change now 👍

Reply:
Thanks we had been trying will.eventually.be.rejected